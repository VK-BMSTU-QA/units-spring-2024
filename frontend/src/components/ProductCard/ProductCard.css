.product-card {
    width: 50vw;
    height: 14vh;

    border-radius: 10px;
    padding: 20px;

    border: 1px solid var(--primary-color);

    display: flex;

    justify-content: space-between;
    align-items: center;

    margin: 5px;
}

.product-card__image {
    max-height: 85%;
}

.product-card__name {
    color: var(--primary-color)
}

.product-card__category {
    color: var(--secondary-color);
}

.product-card__text {
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    justify-content: center;
}

.product-card__price {
    margin: 15px 0;
}