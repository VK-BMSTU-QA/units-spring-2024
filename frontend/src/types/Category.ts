export type Category = 'Электроника' | 'Для дома' | 'Одежда';
