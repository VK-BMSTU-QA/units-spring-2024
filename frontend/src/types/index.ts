export * from './Symbol';
export * from './Product';
export * from './Category';
