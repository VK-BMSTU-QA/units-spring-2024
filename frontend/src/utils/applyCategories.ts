import type { Category, Product } from '../types';

export const applyCategories = (
    products: Product[],
    categories: Category[]
): Product[] => {
    if (!categories.length) return products;

    return products.filter((product) => categories.includes(product.category));
};
