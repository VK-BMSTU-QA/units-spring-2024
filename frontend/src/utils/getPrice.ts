import type { PriceSymbol } from '../types';

export const getPrice = (value: number, symbol: PriceSymbol = 'â‚½'): string =>
    `${value.toLocaleString('ru-RU')} ${symbol}`;

    