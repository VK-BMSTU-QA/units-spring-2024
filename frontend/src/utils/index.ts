export * from './updateCategories';
export * from './getPrice';
export * from './applyCategories';

