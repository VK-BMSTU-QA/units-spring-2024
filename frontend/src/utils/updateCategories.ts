import type { Category } from '../types';

export const updateCategories = (
    currentCategories: Category[],
    changedCategories: Category
) => {
    if (currentCategories.includes(changedCategories)) {
        return currentCategories.filter(
            (filter) => filter !== changedCategories
        );
    }

    return [...currentCategories, changedCategories];
};

