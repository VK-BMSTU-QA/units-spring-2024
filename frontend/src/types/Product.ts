import { Category } from './Category';
import { PriceSymbol } from './Symbol';

export type Product = {
    id: number;
    name: string;
    description: string;
    price: number;
    priceSymbol?: PriceSymbol;
    imgUrl?: string;
    category: Category;
};
