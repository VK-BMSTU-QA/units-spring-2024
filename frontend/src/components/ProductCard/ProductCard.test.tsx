import { render } from '@testing-library/react';
import { ProductCard } from './ProductCard';
import { Product } from '../../types/Product';
import { getPrice } from '../../utils';

const testProduct: Product = {id: 1, name: 'Aspire ES 15', description: 'Ноутбук', price: 100, category: 'Электроника'}

jest.mock('../../utils/getPrice', () => ({
    getPrice: jest.fn()
}));

afterEach(jest.clearAllMocks);

describe('Categories test', () => {
    it('should render correctly', () => {
        const rendered = render(<ProductCard
            {...testProduct}
            priceSymbol='₽'
             />);

        expect(rendered.asFragment()).toMatchSnapshot();
    });
    it('should call getPrice when rendering', () => {
        expect(getPrice).toHaveBeenCalledTimes(0);

        const rendered = render(<ProductCard
            {...testProduct}
            priceSymbol='₽'
             />);

        expect(getPrice).toHaveBeenCalledTimes(1);
    });
});
