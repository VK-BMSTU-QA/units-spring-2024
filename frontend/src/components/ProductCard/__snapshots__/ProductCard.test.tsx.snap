// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Categories test should render correctly 1`] = `
<DocumentFragment>
  <div
    class="product-card"
  >
    <div
      class="product-card__text"
    >
      <h2
        class="product-card__name"
      >
        Aspire ES 15
      </h2>
      <div
        class="product-card__description"
      >
        Ноутбук
      </div>
      <h1
        class="product-card__price"
      />
      <div
        class="product-card__category"
      >
        Электроника
      </div>
    </div>
  </div>
</DocumentFragment>
`;
